### DISCUSSION TITLE
TuringCourses/major_module/principle_of_programming_languages/

### DISCUSSION THREAD

**Discussion by giscus** [ZJU-Turing/TuringCourses#127] [General]
# TuringCourses/major_module/principle_of_programming_languages/

图灵班学习指南

https://zju-turing.github.io/TuringCourses/major_module/principle_of_programming_languages/

**Comment by MTDickens** [comment]
为啥 rainoftime 不来教这节课，感觉 ZJU 最适合教 PLT 的就只有他了 :cry:

**Reply by MTDickens** [reply]
姚老师说目前没有打算开课，原因是：

1. 因为教了编译原理，KPI 也够了
2. 没法给本科生新开一门课
3. 想教一门 workload 比较小的课，同时也许能够吸引一下学生

其实感觉姚老师也许本身就不是很想教这门课。

所以看看今年选课的人数吧。目前已经有 50 多人（一半来自竺院），如果人数够多，说不定姚老师能来教。

**Reply by MTDickens** [reply]
如果你比较感兴趣的话，建议学习 PFPL, TAPL (Types and Programming Languages), ATTPL (Advanced Topics in Types and Programming Languages
)。

- 我最近刚开始看 TAPL，认真读的话，书还是可以写的很清晰的。

另外姚老师推荐：感兴趣的可以去 [Bruno](https://i.cs.hku.hk/~bruno/) 那里暑研/深造。

- Bruno 的链接已更正，感谢下面的大佬

**Reply by floatshadow** [reply]
你把 Bruno 搞混了，应该是 [Bruno C. d. S. Oliveira](https://i.cs.hku.hk/~bruno/)

**Comment by ZelinMa557** [comment]
其实翁老师第一年讲的还是可以的。可能和当时有几位大咖助教把关有关系

**Reply by catturb** [reply]
求问是哪个学年学期的啊，是ppl还是改名后的，想看看这个讲的可以的智云
